overlord/snapstate/snapstate.go

```diff
@@ -969,7 +973,8 @@ func doUpdate(ctx context.Context, st *state.State, names []string, updates []*s
 		snapsup := &SnapSetup{
 			Base:         update.Base,
 			Prereq:       defaultContentPlugProviders(st, update),
-			Channel:      updateOpts.Channel,
+			Channel:      revnoOpts.Channel,
+			CohortKey:    revnoOpts.CohortKey,
```

Comment:
this should be snapst.CohortKey if that revno.CohortKey is "" or the logic ins doLinkSnap etc needs to change

Reply:
it's defaulted to snapst.CohortKey in the params() func

Reply:
sorry, not in `parms()`; search for `// default to being in the same cohort`

Reply:
sigh. Final clarification: for single snap update, it's defaulted to snapst.CohortKey if opts.CohortKey is unset. In multi-snap update, opts.CohortKey is ignored and it's set to snapst.CohortKey.